### DISCUSSION TITLE
Working with documentation from submodules and the repository link

### DISCUSSION THREAD

**Discussion by PlasmaHH** [squidfunk/mkdocs-material#7299] [Q&A]
So this might be a bit of a stretch but maybe someone has already done something similar and knows a good way to deal with this situation.

I pull our full documentation together from various git repos and submodules (and symlink those into my main docs somewhere). Now the mkdocs.yml supports  only the main repo url, which makes sense since we don't really want mkdocs to know about and deal with git details.

So the question mainly could boil down to "How can I set/influence the repository url on a per directory (tree) base?".

I thought about maybe something with the metadata per page and the actions partial? But then I would have to set the meta data per file, there does not seem to be a way to do it for a whole directory...

**Comment by kamilkrzyskow** [comment]
Hello @PlasmaHH,
the repository information in the header is taken from the global `config.repo_url` in the `source.html` template:
- https://github.com/squidfunk/mkdocs-material/blob/12a8e82837fe400dd1f123e41d75b32987a11744/src/templates/partials/source.html#L25

and each of the pages have `page.edit_url` used in the `actions.html` template responsible for the view/edit buttons.
- https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/actions.html

---

So in general you have [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/) options that should be easier to configure than customization like:
- ~~https://squidfunk.github.io/mkdocs-material/plugins/projects/~~
  ~~Separating the config for the subdirectories, which solves the global `config.repo_url` issue, which in turn solves the local `page.edit_url` issue.~~
  EDIT: Actually to transform the URLs it still requires some additional Python based on the [example here](https://github.com/mkdocs-material/examples/blob/master/examples/__init__.py), so I think, in this scenario, setting up a hook will be easier.
- https://squidfunk.github.io/mkdocs-material/plugins/meta/
  Allowing to set `meta` values for whole directories, but then still requires modifying the templates.

---

Apart of those you can use [hooks](https://www.mkdocs.org/user-guide/configuration/#hooks) to edit the meta for files based on the `page.file.src_uri` property + edit the templates to use the values from the meta per each page. Hooks would also allow you to modify the global `config.repo_url`.

You could also implement the logic to set different URLs based on the `page.file.src_uri` directly in the templates without setting them via hooks, but depending on the amount of subpaths the template could become less readable.
